### PR TITLE
fix(developer-guide): tweak developer guide

### PR DESCRIPTION
**Pull request by SamVerschueren**
Some minor tweaks I came across when reading the developer guide. Let me know if you want to revert some things.

Also, not sure about the commit message...

### CODE REVIEW

File: DEVELOPERS_GUIDE.md [hunk 1]
```diff
@@ -261,13 +263,13 @@ To run the server, navigate to the `server` folder and execute the following com
 
 ```
 $ cd server
-$ yarn run build -- --env=YOUR_ENV_NAME && node dist/index.js
+$ yarn build --env=YOUR_ENV_NAME && node dist/index.js
```
Comment (SamVerschueren) [inline DEVELOPERS_GUIDE.md]: `run` is optional here so can be removed. I'm fine with adding it back. The `--` however should be removed. When running it with `yarn@1` it gives this error message

> From Yarn 1.0 onwards, scripts don't require "--" for options to be forwarded. In a future version, any explicit "--" will be forwarded as-is to the scripts

**Comment by SamVerschueren** [comment]
We should also specify explicitely which Node version to use. The server as well as the client both requires Node 8. Any suggestions where we should put it?

File: DEVELOPERS_GUIDE.md (hunk 1 above)
Comment (PascalPrecht) [reply DEVELOPERS_GUIDE.md]: Yes, good catch!

**Comment by PascalPrecht** [comment]
Thanks for the PR @SamVerschueren !

I'm fine with nailing the node version down to current LTS version. While you're at it, would you mind adding yarn as a local dependency all the packages and change the developer guide to use `npx yarn install` instead?

That'd be great.

re: commit message: I'd go with a `docs` type here. So basically:

```
docs(developer-guide): fix yarn command in server docs
```

All following changes however, should be in a separate commit each.

File: DEVELOPERS_GUIDE.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ This guide helps you setting up, building and running MachineLabs on your own ma
 ## Prerequisites
 
 - Get a [Firebase](https://firebase.google.com) account
-- Install [Git](https://git-scm.com) and [NodeJS](https://nodejs.org)
+- Install [Git](https://git-scm.com) and [NodeJS 8+](https://nodejs.org)
```
Comment (SamVerschueren) [inline DEVELOPERS_GUIDE.md]: Is this good enough?

File: DEVELOPERS_GUIDE.md [hunk 3]
```diff
@@ -276,7 +261,7 @@ To run the client, we use Angular CLI, as the client is an Angular app. All we h
 
 ```
 $ cd client
-$ ng serve --env=YOUR_ENV_NAME
+$ ng serve --env=personal
```
Comment (SamVerschueren) [inline DEVELOPERS_GUIDE.md]: See above why this is changed to `personal`.

**Comment by SamVerschueren** [comment]
@d3lm Any idea why the server build fails? Also, if I run `npm run format:base -- --list-different` locally inside the server, I get the following error.

```
$ npm run format:base -- --list-different

> @machinelabs/server@0.28.1 format:base /Users/sam/Projects/opensource/forks/machinelabs/server
> prettier --config ../.prettierrc --ignore-path ../.prettierignore '**/*{.ts,.js,.json}' "--list-different"

package.json
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @machinelabs/server@0.28.1 format:base: `prettier --config ../.prettierrc --ignore-path ../.prettierignore '**/*{.ts,.js,.json}' "--list-different"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @machinelabs/server@0.28.1 format:base script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sam/.npm/_logs/2018-03-23T06_27_13_113Z-debug.log
```

The debug log is not of much help.

File: DEVELOPERS_GUIDE.md (hunk 2 above)
Comment (PascalPrecht) [reply DEVELOPERS_GUIDE.md]: Yes that should do it

File: DEVELOPERS_GUIDE.md [hunk 4]
```diff
@@ -204,28 +204,11 @@ export const environment = {
...
-  "personal": "environments/environment.personal.ts",
-  "offline": "environments/environment.offline.ts",
-  "staging": "environments/environment.staging.ts",
-  "production": "environments/environment.production.ts",
-  "test": "environments/environment.test.ts",
-  "e2e": "environments/environment.e2e.ts"
-}
-...
-```
+We can repeat the process for the client. Create a file `environment.personal.ts` in `client/src/environments`.
```
[the remark this answers is not in the document]
Comment (PascalPrecht) [reply DEVELOPERS_GUIDE.md]: Yes, true. And I'm fine with taking this change. However, it's important that people understand they aren't limited to just using `environment.personal.ts` but could basically come up with whatever suits them.

File: DEVELOPERS_GUIDE.md [hunk 5]
```diff
@@ -243,6 +226,8 @@ export const environment = {
 }
 ```
 
+Make sure to replace the `firebaseConfig` with your own. You can retrieve the firebase configuration from your firebase project by going to **Project Overview -> Add Firebase to your web app**.
```
Comment (PascalPrecht) [inline DEVELOPERS_GUIDE.md]: 👍

File: DEVELOPERS_GUIDE.md (hunk 3 above)
Comment (PascalPrecht) [reply DEVELOPERS_GUIDE.md]: 👍

File: DEVELOPERS_GUIDE.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (SamVerschueren) [reply DEVELOPERS_GUIDE.md]: So you want me to add a small extra note regarding adding a custom `environment.[NAME].ts` file as well?

File: DEVELOPERS_GUIDE.md [hunk 6]
```diff
@@ -204,28 +204,11 @@ export const environment = {
...
-  "personal": "environments/environment.personal.ts",
-  "offline": "environments/environment.offline.ts",
-  "staging": "environments/environment.staging.ts",
-  "production": "environments/environment.production.ts",
-  "test": "environments/environment.test.ts",
-  "e2e": "environments/environment.e2e.ts"
-}
-...
-```
+We can repeat the process for the client. Create a file `environment.personal.ts` in `client/src/environments`. This file is ignored by source control. Feel free to add other environment files by creating a `environment.[NAME].ts` file and adding an entry in [angular-cli.json](https://github.com/angular/angular-cli/wiki/build#build-targets-and-environment-files) accordingly.
```
Comment (SamVerschueren) [inline DEVELOPERS_GUIDE.md]: @PascalPrecht can you check this one again? If you like to have it differently, let me know and I'll change it.

**Comment by SamVerschueren** [comment]
Something that comes in my mind that's undocumented right now. When running the server locally, it's necessary to have the Google Cloud SDK installed. Is this something that we should also cover? Could be a separate PR from this. Just wanted to ask.

File: DEVELOPERS_GUIDE.md (hunk 6 above)
Comment (PascalPrecht) [reply DEVELOPERS_GUIDE.md]: This LGTM

**Comment by PascalPrecht** [comment]
This is already covered in the developer guide under "Prerequisites" https://github.com/machinelabs/machinelabs/blob/master/DEVELOPERS_GUIDE.md#prerequisites

**Comment by PascalPrecht** [comment]
LGTM. Testing the local yarn deps now.

**Comment by PascalPrecht** [comment]
Alright, everything seems to work and travis is happy as well. Merging.

**Comment by SamVerschueren** [comment]
Ah totally missed the GCloud prerequisite. Awesome! Thanks for reviewing and merging!

**Pull request merged**